articles/142_idl.md

```diff
@@ -35,3 +35,45 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Type Mapping
+
+IDL types mapped to different types in different languages.
+
+
+| IDL type | C type    | C++ type    | Python type          |
+| -------- | --------- | ----------- | -------------------- |
+| bool     | \_Bool    | bool        | bool                 |
+| octet    | uint8\_t  | uint8\_t    | bytes with length 1  |
+| char     | char      | char        | str with length 1    |
```

Comment:
In C char type maps to signed char: https://github.com/ros2/rosidl/blob/ac4ac643d55cd7fb6c9604d0e67dc86dcd8d8d53/rosidl_generator_c/include/rosidl_generator_c/primitives_array.h#L33

Reply:
As this is a string, the question of the encoding may propagate here as well

Reply:
7c1d4a45826ffbb9137305e1c8848d69697b10a0 for signed char